app/scripts/controllers/swaps.js

```diff
@@ -685,34 +723,51 @@ export default class SwapsController {
 }
 
 /**
- * Calculates the median of a sample of BigNumber values.
+ * Calculates the median overallValueOfQuote of a sample of quotes.
  *
- * @param {import('bignumber.js').BigNumber[]} values - A sample of BigNumber
- * values. The array will be sorted in place.
- * @returns {import('bignumber.js').BigNumber} The median of the sample.
+ * @param {array} quotes - A sample of quote objects with overallValueOfQuote and ethFee properties
+ * @returns {object} An object with the media overallValueOfQuote for all quotes, and the associated median ethFee and metaMaskFeeInEth.
  */
-function getMedian(values) {
-  if (!Array.isArray(values) || values.length === 0) {
+function getMedianEthValueQuote(_quotes) {
+  if (!Array.isArray(_quotes) || _quotes.length === 0) {
     throw new Error('Expected non-empty array param.')
   }
 
-  values.sort((a, b) => {
-    if (a.equals(b)) {
+  const quotes = [..._quotes]
+
+  quotes.sort((quoteA, quoteB) => {
+    const overallValueOfQuoteA = new BigNumber(quoteA.overallValueOfQuote, 10)
+    const overallValueOfQuoteB = new BigNumber(quoteB.overallValueOfQuote, 10)
+    if (overallValueOfQuoteA.equals(overallValueOfQuoteB)) {
       return 0
     }
-    return a.lessThan(b) ? -1 : 1
+    return overallValueOfQuoteA.lessThan(overallValueOfQuoteB) ? -1 : 1
   })
 
-  if (values.length % 2 === 1) {
+  if (quotes.length % 2 === 1) {
     // return middle value
-    return values[(values.length - 1) / 2]
+    return quotes[(quotes.length - 1) / 2]
   }
 
   // return mean of middle two values
-  const upperIndex = values.length / 2
-  return values[upperIndex].plus(values[upperIndex - 1]).dividedBy(2)
+  const upperIndex = quotes.length / 2
+
+  return {
+    ethFee: new BigNumber(quotes[upperIndex].ethFee, 10)
+      .plus(quotes[upperIndex - 1].ethFee)
```

Comment:
This is missing the radix:

```suggestion
      .plus(quotes[upperIndex - 1].ethFee, 10)
```

Reply:
The two `.plus` calls below are missing the radix as well

Reply:
Done in 1458ad7d1